R/reactives.R

```diff
@@ -682,18 +905,26 @@ srcrefToLabel <- function(srcref, defaultLabel) {
 }
 
 #' @export
+format.reactiveExpr <- function(x, ...) {
+  fun <- attr(x, "observable", exact = TRUE)$.origFunc
+  label <- sprintf('reactive(%s)', paste(deparse(body(fun)), collapse='\n'))
```

Comment:
Instead of reaching into the Observable object for `.origFunc` and then deparsing it, you could add a method to Observable that returns the text.